tests/php/test-amp-helper-functions.php

```diff
@@ -628,6 +629,33 @@ public function test_amp_get_content_embed_handlers() {
 		$this->assertEquals( $post, $this->last_filter_call['args'][1] );
 	}
 
+	/**
+	 * Test amp_is_dev_mode().
+	 *
+	 * @covers ::amp_is_dev_mode()
+	 */
+	public function test_amp_is_dev_mode() {
+		$this->assertFalse( amp_is_dev_mode() );
+		add_filter( 'amp_dev_mode_enabled', '__return_true' );
+		$this->assertTrue( amp_is_dev_mode() );
+		remove_filter( 'amp_dev_mode_enabled', '__return_true' );
+		$this->assertFalse( amp_is_dev_mode() );
+
+		// Test authenticated user with admin bar showing.
+		add_filter( 'show_admin_bar', '__return_true' );
+		wp_set_current_user( self::factory()->user->create( [ 'role' => 'administrator' ] ) );
+		$this->assertTrue( is_admin_bar_showing() );
+		$this->assertTrue( is_user_logged_in() );
+		$this->assertTrue( amp_is_dev_mode() );
+
+		// Test unauthenticated user with admin bar forced.
+		add_filter( 'show_admin_bar', '__return_true' );
```

Comment:
Theoretical issue: The previous `'show_admin_bar'` filter was not removed. As it is the same one, this is a non-issue, but if it would be a different one, or something would be interjected in-between, it could change the test results.

Reply:
Removed prior filter in d79be8df.